plugins/woocommerce/client/admin/client/settings-payments/onboarding/providers/woopayments/steps/business-verification/steps/embedded-kyc.tsx

```diff
@@ -0,0 +1,140 @@
+/**
+ * External dependencies
+ */
+import React, { useEffect, useState } from 'react';
+import { __ } from '@wordpress/i18n';
+import { LoadError } from '@stripe/connect-js';
+
+/**
+ * Internal dependencies
+ */
+import { useOnboardingContext } from '../../../data/onboarding-context';
+import StripeSpinner from '../../../components/stripe-spinner';
+import BannerNotice from '../../../components/banner-notice';
+import { useBusinessVerificationContext } from '../data/business-verification-context';
+import { finalizeOnboarding, isPoEligible } from '../utils';
+import { EmbeddedAccountOnboarding } from '../components/embedded';
+
+interface Props {
+	continueKyc?: boolean;
+	collectPayoutRequirements?: boolean;
+}
+
+const EmbeddedKyc: React.FC< Props > = ( {
+	continueKyc = false,
+	collectPayoutRequirements = false,
+} ) => {
+	const { data } = useBusinessVerificationContext();
+	const { currentStep, navigateToNextStep } = useOnboardingContext();
+	const [ finalizingAccount, setFinalizingAccount ] = useState( false );
+	const [ isEligible, setIsEligible ] = useState< boolean | null >( null );
+	const [ loading, setLoading ] = useState( true );
+	const [ loadError, setLoadError ] = useState< LoadError | null >( null );
+	const fallbackUrl = currentStep?.actions?.kyc_fallback?.href ?? '';
+
+	// Fetch whether the account is eligible for progressive onboarding
+	useEffect( () => {
+		const checkEligibility = async () => {
+			const eligibility = await isPoEligible( data );
+			setIsEligible( eligibility );
+		};
+
+		if ( ! continueKyc ) {
+			checkEligibility();
+		} else {
+			setIsEligible( false );
+		}
+	}, [ continueKyc, data ] );
+
+	const handleStepChange = () => {
+		// To-Do: Track step change.
+	};
+
+	const handleOnExit = async () => {
+		setFinalizingAccount( true );
+
+		try {
+			const response = await finalizeOnboarding(
+				currentStep?.actions?.kyc_session_finish?.href ?? '',
+				'NOX' // To-Do: Replace with the correct source.
```

Comment:
@vladolaru What would be the correct source?

Reply:
@mordeth the supported sources are [here](https://github.com/Automattic/woocommerce-payments/blob/0024a37df240a6c9cb60deb32f145d4fb596ab99/includes/class-wc-payments-onboarding-service.php#L30-L30). But best not to send anything if you don't have reliable data on hand - it is up to the backend to fill in the gaps.

Reply:
Could we not use `wcadmin-settings-page` here? Or would we want to add a new source, specifically for NOX?